docs/observability.md

```diff
@@ -12,6 +12,101 @@ If you need to troubleshoot the Collector, see [Troubleshooting].
 Read on to learn about experimental features and the project's overall vision
 for internal telemetry.
 
+<!-- toc -->
+
+- [Goals of internal telemetry](#goals-of-internal-telemetry)
+  * [Observable elements](#observable-elements)
+  * [Impact](#impact)
+  * [Configurable level of observability](#configurable-level-of-observability)
+  * [Internal telemetry properties](#internal-telemetry-properties)
+    + [Units](#units)
+    + [Process for defining new metrics](#process-for-defining-new-metrics)
+- [Experimental trace telemetry](#experimental-trace-telemetry)
+
+<!-- tocstop -->
+
+## Goals of internal telemetry
+
+The Collector's internal telemetry is an important part of fulfilling
+OpenTelemetry's [project vision](vision.md). The following section explains the
+priorities for making the Collector an observable service.
+
+### Observable elements
+
+The following aspects of the Collector need to be observable.
+
+- [Current values]
+  - Some of the current values and rates might be calculated as derivatives of
+    cumulative values in the backend, so it's an open question whether to expose
+    them separately or not.
+- [Cumulative values]
+- [Trace or log events]
+  - For start or stop events, an appropriate hysteresis must be defined to avoid
+    generating too many events. Note that start and stop events can't be
+    detected in the backend simply as derivatives of current rates. The events
+    include additional data that is not present in the current value.
+- [Host metrics]
+  - Host metrics can help users determine if the observed problem in a service
+    is caused by a different process on the same host.
+
+### Impact
+
+The impact of these observability improvements on the core performance of the
+Collector must be assessed.
+
+### Configurable level of observability
+
+Some metrics and traces can be high volume and users might not always want to
+observe them. An observability verboseness “level” allows configuration of the
+Collector to send more or less observability data or with even finer
+granularity, to allow turning on or off specific metrics.
+
+The default level of observability must be defined in a way that has
+insignificant performance impact on the service.
+
+### Internal telemetry properties
+
+Telemetry produced by the Collector has the following properties:
+
+- metrics produced by Collector components use the prefix `otelcol_`
+- metrics produced by any instrumentation library used by Collector components will *not* be prefixed with `otelcol_`
+- code is instrumented using the OpenTelemetry API and telemetry is produced via the OpenTelemetry Go SDK
```

Comment:
The API part is technically not true for logs, right? (We would use zap with a bridge?) Not sure if that matters much, maybe we just need to mention the SDK and not the API part

Reply:
I wanted to capture this for any component author asking the question "how should i instrument my code". I've updated the text to reflect that we use zap for logging.